reindex.go

```diff
@@ -287,13 +287,20 @@ func ReindexPrefix(db *ICATConnection, es *ESConnection, prefix, irodsZone strin
 	start := time.Now()
 	defer logTime(prefixlog, start, &rows)
 
-	tx, err := db.BeginTx(context.TODO(), nil)
+	seenEsDocs := make(map[string]bool)
+	docs, esDocs, esDocTypes, err := getSearchResults(prefixlog, prefix, es)
+	rows.documents = docs
+	if err != nil {
+		return err
+	}
+
+	tx, err := db.BeginTx(context.Background(), nil)
```

Comment:
Basically I just swapped these, since the search results part doesn't need access to the DB transaction.